src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,421 @@
+package org.broadinstitute.hellbender.tools.walkers.genotyper;
+
+import htsjdk.variant.variantcontext.*;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.broadinstitute.hellbender.utils.variant.GATKVariantContextUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Created by davidben on 6/8/16.
+ */
+public class AlleleSubsettingUtils {
+    private AlleleSubsettingUtils() {}  //prevent instantiation
+    private static final int PL_INDEX_OF_HOM_REF = 0;
+    private static final int MAX_LENGTH_FOR_PL_LOGGING = 100; // if PL vectors longer than this # of elements, don't log them
+
+    /**
+     * From a given variant context, extract a given subset of alleles, and update genotype context accordingly,
+     * including updating the PL's, and assign genotypes accordingly
+     * @param vc                                variant context with alleles and genotype likelihoods
+     * @param defaultPloidy                     ploidy to use when {@code vc} is missing information for a sample
+     * @param allelesToKeep                     subset of alleles in {@code vc} to keep
+     * @param method                            whether to assign hard genotypes or leave as no-call
+     * @return                                  GenotypesContext with new PLs
+     */
+    public static GenotypesContext subsetAlleles(final VariantContext vc, final int defaultPloidy, final boolean isDiploid,
```

Comment:
Instead of having an argument to allow the user to tell you is diploid, could you just check whether it is indeed diploid. If more efficiency you want to allow the user to by pass such a check you could make ```subsetDiploidAlleles``` public; the latter method should fail gracefully (IAE) if encounters something that is not diploid.